MSAL/test/unit/native_auth/public/delegate/sign_in/SignInStartDelegateDispatcherTests.swift

```diff
@@ -0,0 +1,151 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+@testable import MSAL
+import XCTest
+
+final class SignInStartDelegateDispatcherTests: XCTestCase {
+    private var telemetryExp: XCTestExpectation!
+    private var delegateExp: XCTestExpectation!
+    private var sut: SignInStartDelegateDispatcher!
+    private let controllerFactoryMock = MSALNativeAuthControllerFactoryMock()
+    private let correlationId = UUID()
+
+    override func setUp() {
+        super.setUp()
+        telemetryExp = expectation(description: "delegateDispatcher telemetry exp")
+        delegateExp = expectation(description: "delegateDispatcher delegate exp")
+    }
+
+    func test_dispatchSignInCodeRequired_whenDelegateMethodsAreImplemented() async {
+        let expectedState = SignInCodeRequiredState(scopes: [], controller: controllerFactoryMock.signInController, flowToken: "flowToken", correlationId: correlationId)
+        let expectedSentTo = "user@contoso.com"
+        let expectedChannelTargetType = MSALNativeAuthChannelType.email
+        let expectedCodeLength = 4
+
+        let delegate = SignInCodeStartDelegateSpy(expectation: delegateExp, expectedSentTo: expectedSentTo, expectedChannelTargetType: expectedChannelTargetType, expectedCodeLength: expectedCodeLength)
+
+        sut = .init(delegate: delegate, telemetryUpdate: { result in
+            guard case .success = result else {
+                return XCTFail("wrong result")
+            }
+            self.telemetryExp.fulfill()
+        })
+
+        await sut.dispatchSignInCodeRequired(
+            newState: expectedState,
+            sentTo: expectedSentTo,
+            channelTargetType: expectedChannelTargetType,
+            codeLength: expectedCodeLength
+        )
+
+        await fulfillment(of: [telemetryExp, delegateExp])
+
+        XCTAssertEqual(delegate.expectedState, expectedState)
+        XCTAssertEqual(delegate.expectedSentTo, expectedSentTo)
+        XCTAssertEqual(delegate.expectedChannelTargetType, expectedChannelTargetType)
+        XCTAssertEqual(delegate.expectedCodeLength, expectedCodeLength)
+    }
+
+    func test_dispatchSignUpCodeRequired_whenDelegateOptionalMethodsNotImplemented() async {
```

Comment:
This should be `test_dispatchSignInCodeRequired_whenDelegateOptionalMethodsNotImplemented` not `test_dispatchSignUpCodeRequired_whenDelegateOptionalMethodsNotImplemented` SignIn vs SignUp

Reply:
Thanks, changed.